pkg/k8sutil/create.go

```diff
@@ -161,3 +168,60 @@ func parseJSONManifest(data []byte) ([]manifest, error) {
 	}
 	return manifests, nil
 }
+
+// CreateOrUpdateNamespace creates the release namespace or updates the namespace
+// if it already exists.
+func CreateOrUpdateNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) error {
```

Comment:
Hm, `Get` + `Create` Or `Update` would be slightly more optimal than `Create` + `Get` + `Update`.

We will run more updates and creation only once, but I'm not sure which one should be prioritized.

Or just run all 3 operations all the time to make it fixed :smile:

Reply:
made the change to first get the namespace, if it doesnt exist create , else update.
Also added a retry logic in case of conflict that might come if the namespace is simultaneously updated elsewhere.